app/models/piece.rb

```diff
@@ -1,3 +1,73 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def exist?(x, y)
+    Piece.exists?(x_position: x, y_position: y)
```

Comment:
Looking good!!!!!!

Instead of using this method here, I suggest using active record directly, like this:

`game.pieces.exists?(x_position: x, y_position: y)`

The important part is, see how I'm using `game.pieces`. This ensures we only look at pieces within the current game.

The current code will look at pieces from ALL games in our database.